internal/engine/actions/alert/alert.go

```diff
@@ -43,7 +43,11 @@ func NewRuleAlert(rt *pb.RuleType, pbuild *providers.ProviderBuilder) (engif.Act
 		if alertCfg.GetSecurityAdvisory() == nil {
 			return nil, fmt.Errorf("alert engine missing security-advisory configuration")
 		}
-		return security_advisory.NewSecurityAdvisoryAlert(ActionType, rt.GetSeverity(), alertCfg.GetSecurityAdvisory(), pbuild)
+		client, err := pbuild.GetGitHub()
```

Comment:
Can we first check if the provider implements github and skip if it doesn't?

Reply:
I'm not sure I agree with this suggestion:

1. The code copies the existing behaviour: https://github.com/stacklok/minder/blob/main/internal/engine/actions/alert/security_advisory/security_advisory.go#L161
2. If other aspects of the configuration are incorrect, we also return an error.

Reply:
Note in my next PR, I will change the behaviour to print an explicit error if the provider is not Github

Reply:
There is an explicit "error" that gets the engine to mark something as skipped. Anyway, it's fine to do it in a separate PR.